Source/WebCore/platform/graphics/gstreamer/mse/PlaybackPipeline.cpp

```diff
@@ -381,6 +373,10 @@ void PlaybackPipeline::flush(AtomString trackId)
 
     if (!gst_base_src_new_seamless_segment(GST_BASE_SRC(appsrc), segment->start, segment->stop, segment->start)) {
         GST_WARNING("Failed to send seamless segment event for trackId=%s", trackId.string().utf8().data());
+    }
+
+    if (!gst_element_send_event(GST_ELEMENT(appsrc), gst_event_new_flush_stop(false))) {
```

Comment:
This doesn't look right. The order of emission of the flush-start, flush-stop and segment events is [very specific in GStreamer](https://gstreamer.freedesktop.org/documentation/plugin-development/advanced/events.html?gi-language=c#flush-stop). You can't just swap the order. If you do, elements might just not be prepared to receive the segment event, or will forget it as soon as flush-stop arrives.

The current code already allows (or should allow, unless there's some bug) independent flush of audio/video streams. The way it works is this: we ask the current position to the sink (to know the last frame rendered), we flush everything (the main reason is to flush the decoder), send a new segment configured for playback from the last rendered position (although earlier buffers may be enqueued, because we must enqueue since the last key frame to allow the decoder do its work properly), and then start to enqueue the relevant data.

Reply:
gst_base_src_new_seamless_segment doesn't send the event (see gstbasesrc.c). it configures one to be sent after streaming is restarted after flush_stop. the order of events doesn't change. 

it, however, fixes a race between a streaming thread (restarted on flush_stop) and gst_base_src_new_seamless_segment trying to set new segment. 

> The current code already allows (or should allow, unless there's some bug) independent flush of audio/video streams.

I think the webkitMediaSrcChain was introduced after flush changes. and flow combiner may result in audio streaming thread to stop because of GST_FLOW_FLUSHING returned from video flush...